api/credentialvalidator/credentialvalidator.go

```diff
@@ -70,3 +70,17 @@ func (c *Facade) WatchCredential(credentialID string) (watcher.NotifyWatcher, er
 	w := apiwatcher.NewNotifyWatcher(c.facade.RawAPICaller(), result)
 	return w, nil
 }
+
+// InvalidateModelCredential invalidates cloud credential for the model that made a connection.
+func (c *Facade) InvalidateModelCredential(reason string) error {
+	var result params.ErrorResult
+	err := c.facade.FacadeCall("InvalidateModelCredential", reason, &result)
```

Comment:
Interesting. I would have expected with a single return value, there's no need for an ErrorResult struct at all. Just return the err from the function. But I see 3 or 4 cases where we also do it this way.

Reply:
I seem to recall that this is the infrastructure thing - we need to return something and an error. So it makes no sense to return error, error. Hence, ErrorResult, error is returned.
And yes, a few places had to follow this to fit the pattern.